core/src/main/scala/com/dwolla/scheduledmaintenance/Main.scala

```diff
@@ -2,24 +2,32 @@ package com.dwolla.scheduledmaintenance
 
 import dev.holt.javatime.literals.offsetDateTime
 import io.circe.literal._
-import org.scalajs.dom.{FetchEvent, Request, Response, ResponseInit}
+import org.scalajs.dom._
 import stubs.Globals
 
 import java.time.format.DateTimeFormatter
 import java.time.{OffsetDateTime, ZoneOffset}
+import scala.annotation.nowarn
+import scala.concurrent.ExecutionContext.Implicits.global
+import scala.concurrent.Future
 import scala.scalajs.js
+import scala.scalajs.js.JSConverters.JSRichFutureNonThenable
 
 object Main {
+  @nowarn("cat=other")
   def main(args: Array[String]): Unit =
     Globals.addEventListener("fetch",
-      (event: FetchEvent) => event.respondWith(handleRequest(event.request))
+      (event: FetchEvent) => event.respondWith(handleRequest(event.request).toJSPromise)
     )
 
   //noinspection SameParameterValue
   private def formatForHttpHeader(odt: OffsetDateTime): String =
     odt.toInstant.atOffset(ZoneOffset.UTC).format(DateTimeFormatter.RFC_1123_DATE_TIME)
 
-  private[scheduledmaintenance] def handleRequest(req: Request): Response =
+  private[scheduledmaintenance] def handleRequest(req: Request): Future[Response] = {
+    if(Option(req.headers.get("Access")).exists(_.contains("<secret-code>")))
+      return fetch(req).toFuture
```

Comment:
Is this going to allow the request to pass through if the Access header matches a secret?

Reply:
no, this will allow the request to pass through if the Access header contains the exact string `<secret-code>`. Figured that was fine for a sandbox test. We can do something more sophisticated for uat and prod!